Lesson25/selcan/Lesson25/filter.js

```diff
@@ -0,0 +1,410 @@
+	'With each step, the hiker felt closer to the summit, fueled by determination and a sense of adventure.',
+	'The aroma of freshly brewed coffee permeated the air, enticing passersby to indulge in a cup of warmth.',
+	'The butterfly gracefully fluttered its wings, showcasing vibrant colors in a delicate dance.',
+];
+console.log("Exercise5:",[...sentencesExercise5].filter(item => item.split(" ").length<=10))
+
+
+// Exercise 6:
+// Given an array of products, filter out the products that are out of stock (where the quantity is 0).
+const productsExercise6 = [
+	{
+		name: 'Laptop',
+		brand: 'Apple',
+		price: 1999,
+		quantity: 0,
+	},
+	{
+		name: 'Smartphone',
+		brand: 'Samsung',
+		price: 899,
+		quantity: 20,
+	},
+	{
+		name: 'TV',
+		brand: 'LG',
+		price: 1499,
+		quantity: 15,
+	},
+	{
+		name: 'Headphones',
+		brand: 'Sony',
+		price: 299,
+		quantity: 30,
+	},
+	{
+		name: 'Smartwatch',
+		brand: 'Apple',
+		price: 399,
+		quantity: 25,
+	},
+	{
+		name: 'Camera',
+		brand: 'Canon',
+		price: 999,
+		quantity: 0,
+	},
+	{
+		name: 'Gaming Console',
+		brand: 'Microsoft',
+		price: 499,
+		quantity: 8,
+	},
+	{
+		name: 'Tablet',
+		brand: 'Samsung',
+		price: 649,
+		quantity: 18,
+	},
+	{
+		name: 'Smart Speaker',
+		brand: 'Amazon',
+		price: 99,
+		quantity: 0,
+	},
+	{
+		name: 'Fitness Tracker',
+		brand: 'Fitbit',
+		price: 129,
+		quantity: 14,
+	},
+];
+console.log("Exercise6:",productsExercise6.filter(item => item.quantity>0))
+
+// Exercise 7:
+// Given an array of numbers, filter out the numbers that are divisible by 3.
+const numbersExercise7 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+ 
+console.log("Exercise7:",numbersExercise7.filter(item =>item %3 ===0))
+
+// Exercise 8:
+// Given an array of strings, filter out the strings that have more than 5 characters.
+const wordsExercise8 = [
+	'apple',
+	'banana',
+	'car',
+	'dog',
+	'elephant',
+	'flower',
+	'guitar',
+	'house',
+	'island',
+	'jungle',
+];
+console.log("Exercise8:",wordsExercise8.filter(item =>item.length <=4))
+
+// Exercise 9:
+// Given an array of objects representing students, filter out the students who have a grade lower than C.
+const studentsExercise9 = [
+	{
+		id: 1,
+		name: 'Alice',
+		age: 18,
+		grade: 'A',
+	},
+	{
+		id: 2,
+		name: 'Bob',
+		age: 19,
+		grade: 'B',
+	},
+	{
+		id: 3,
+		name: 'Charlie',
+		age: 20,
+		grade: 'B+',
+	},
+	{
+		id: 4,
+		name: 'David',
+		age: 19,
+		grade: 'A-',
+	},
+	{
+		id: 5,
+		name: 'Emma',
+		age: 18,
+		grade: 'A+',
+	},
+	{
+		id: 6,
+		name: 'Frank',
+		age: 20,
+		grade: 'B',
+	},
+	{
+		id: 7,
+		name: 'Grace',
+		age: 19,
+		grade: 'D+',
+	},
+	{
+		id: 8,
+		name: 'Hannah',
+		age: 20,
+		grade: 'A-',
+	},
+	{
+		id: 9,
+		name: 'Isaac',
+		age: 18,
+		grade: 'D',
+	},
+	{
+		id: 10,
+		name: 'Julia',
+		age: 19,
+		grade: 'B-',
+	},
+	]
+	console.log("Exercise9:",studentsExercise9.filter(student => student.grade === "D" || student.grade === "F" ||student.grade === "D+")) ;
```

Comment:
To filter out students with grades lower than C you need to filter out the following greades: C-, D+, D-, D, E+, E, E-, F+, F, F-.
```suggestion
	console.log("Exercise9:",studentsExercise9.filter(student => student.grade[0].toUpperCase() !== "D" || student.grade[0].toUpperCase() !== "F" || student.grade[0].toUpperCase() !== "D"));
```